### PR TITLE
Changes 

### PR DESCRIPTION
**Pull request by Incompleteusern**
- Fixed dialogue being cancelled, see https://discord.com/channels/394189072635133952/709413517907722300/888624905543909457
- SkillPointOverlay changes image
- Added some InfoFormatters and removed others in https://github.com/Wynntils/Wynntils/pull/349/commits/ca94b927c164f548a608a18d55f4efeb18f1613c
- ItemSpecificationOverlay now works on cursor, can nest ScreenRenderers
image

### CODE REVIEW

**Review by magicus — APPROVED** [review]
I think it lokos good. But in the future, please separate large refactorings like the ScreenRenderer change, into a separate PR. That makes reviewing much easier.

File: build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
         maven { url = "http://maven.minecraftforge.net" }
     }
     dependencies {
-        classpath 'net.minecraftforge.gradle:ForgeGradle:4.1.+'
```
Comment (Incompleteusern) [inline build.gradle]: I apologize for not separating the commits, that would've been easier to review.

Also, should this change be reverted?

Comment (magicus) [reply build.gradle]: I can't say. Was there a reason for fixing ForgeGradle at 4.1.14?

Comment (Incompleteusern) [reply build.gradle]: HeyZeer0 told me to do this when gradle failed at a step

Comment (magicus) [reply build.gradle]: Well keep it then :)

**Comment by Incompleteusern** [comment]
This is now dependent on pr 353

**Comment by Incompleteusern** [comment]
@HeyZeer0 can I have feedback when you have the time?

**Comment by nerd** [comment]
i tried the build attached with the artifact for this commit. it crashed when i attempted to change map markers and didn't render the settings properly. also obv bad form to make a mega PR like this but yeah

**Comment by Incompleteusern** [comment]
Mind giving the crash log? Also, what part of the settings render incorrectly?

**Comment by nerd** [comment]
> Mind giving the crash log? 

yeah, my bad: [crash-2021-10-17_23.28.29-client.txt](https://github.com/Wynntils/Wynntils/files/7367999/crash-2021-10-17_23.28.29-client.txt)

> Also, what part of the settings render incorrectly?

**Comment by Incompleteusern** [comment]
Ill take a look at it. Thanks!

**Comment by nerd** [comment]
finally got around to trying this build again. still broken it seems, got this trying to open the map. [crash-2021-11-01_09.37.07-client.txt](https://github.com/Wynntils/Wynntils/files/7453607/crash-2021-11-01_09.37.07-client.txt)

**Comment by magicus** [comment]
Relevant part of @nerd's crash:
```
java.lang.UnsupportedOperationException: Not valid for MapLabel
	at com.wynntils.modules.map.overlays.objects.MapLabel.renderAt(MapLabel.java:77)
	at com.wynntils.modules.map.overlays.objects.WorldMapIcon.drawScreen(WorldMapIcon.java:84)
	at com.wynntils.modules.map.overlays.ui.WorldMapUI.lambda$drawIcons$4(WorldMapUI.java:317)
	at java.util.ArrayList.forEach(Unknown Source)
	at com.wynntils.modules.map.overlays.ui.WorldMapUI.forEachIcon(WorldMapUI.java:182)
	at com.wynntils.modules.map.overlays.ui.WorldMapUI.drawIcons(WorldMapUI.java:311)
	at com.wynntils.modules.map.overlays.ui.MainWorldMapUI.func_73863_a(MainWorldMapUI.java:153)
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:396)
```

**Comment by magicus** [comment]
@Incompleteusern All these crashes are making me nervous, and a bit wary of this entire patch. I think it would be better if you did the large refactoring separately and checked that in first. Then the "actual" changes from this PR will be smaller and easier to review to make sure there are no more latent crashes.

**Comment by Incompleteusern** [comment]
I regret making the large ScreenRenderer changes in the same pr as this, so Ill try moving the changes to a new branch when I have the time 

EDIT: I've been unable to separate the changes of this pr, but if someone can I am more than willing to accept it.

**Pull request merged**